drv/transceivers-server/src/main.rs

```diff
@@ -253,28 +255,24 @@ impl ServerImpl {
 
         #[derive(Copy, Clone, FromBytes, AsBytes)]
         #[repr(C)]
-        struct StatusAndTemperature {
-            status: u8,
+        struct Temperature {
             temperature: zerocopy::I16<zerocopy::BigEndian>,
         }
 
-        loop {
-            let mut out = StatusAndTemperature::new_zeroed();
-            self.transceivers
-                .get_i2c_status_and_read_buffer(port, out.as_bytes_mut())?;
-            if out.status & Reg::QSFP::PORT0_STATUS::BUSY == 0 {
-                if out.status & Reg::QSFP::PORT0_STATUS::ERROR != 0 {
-                    return Err(FpgaError::ImplError(out.status));
-                } else {
-                    // "Internally measured free side device temperatures are
-                    // represented as a 16-bit signed twos complement value in
-                    // increments of 1/256 degrees Celsius"
-                    //
-                    // - SFF-8636 rev 2.10a, Section 6.2.4
-                    return Ok(Celsius(out.temperature.get() as f32 / 256.0));
-                }
-            }
-            userlib::hl::sleep_for(1);
+        let mut out = Temperature::new_zeroed();
+        let status = self
+            .transceivers
+            .get_i2c_status_and_read_buffer(port, out.as_bytes_mut())?;
+
+        if status.error == FpgaI2CFailure::NoError {
+            // "Internally measured free side device temperatures are
+            // represented as a 16-bit signed twos complement value in
+            // increments of 1/256 degrees Celsius"
+            //
+            // - SFF-8636 rev 2.10a, Section 6.2.4
+            return Ok(Celsius(out.temperature.get() as f32 / 256.0));
+        } else {
+            return Err(FpgaError::ImplError(status.as_bytes()[0]));
```

Comment:
since there's no longer a loop here , the `return` keyword is no longer necessary, and clippy is complaining about it: https://github.com/oxidecomputer/hubris/actions/runs/9162854238/job/25190734510?pr=1787

```suggestion
            Ok(Celsius(out.temperature.get() as f32 / 256.0))
        } else {
            Err(FpgaError::ImplError(status.as_bytes()[0]))
```

Reply:
I have a nasty habit of forgetting to run clippy locally. Thanks for pointing this out as well!